### PR TITLE
Husky prepush

### PR DESCRIPTION
**Pull request by nortonprojects**
## Brief Description

Added testing locally and with Github Actions as an automated step.

Instead of running them at commit-time, when code may not be ready for tests, it runs them at push-time, so it's using theoretically complete code. And it runs both locally and remotely for faster feedback and better enforcement.

## JIRA Link

N/A

## Motivation and Context

Testing is currently a manual step that can be automated

## Types of changes

-   [ ] Bug fix
-   [X] New feature
-   [ ] Breaking change

## Checklist

-   [X] My code follows the code style of this project.
-   [ ] My change requires a change to the documentation.
-   [ ] I have added tests to cover my changes.
-   [X] All new and existing tests passed.

### CODE REVIEW

File: .github/workflows/pr-test.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        steps:
+            - uses: actions/checkout@v2
+
+            - name: Install modules
+              run: npm ci
+
+            - name: Setup Node.js environment
+              uses: actions/setup-node@v2.3.0
+
+            - name: Run Jest Tests
```
Comment (Cheshire89) [inline .github/workflows/pr-test.yml:21]: @markacianfrani didn't we just update testing framework to use `npm run cy.test` ?

**Review by Cheshire89 — CHANGES_REQUESTED** [review]

File: .github/workflows/pr-test.yml (hunk 1 above)
Comment (markacianfrani) [reply .github/workflows/pr-test.yml:21]: For E2E tests. The jest unit tests are still just npm run test

**Comment by markacianfrani** [comment]
I feel like the pre push hook might be too much. I'm thinking of scenarios like resolving a package.json merge conflict or something small. Having to wait for the whole test suite to run could be annoying when I've already just ran it locally. They'll take longer and longer to run too as the project grows. We can just do git push --iforget though so it's not a huge deal. I'm down for trying it. 

The GitHub action is perfect though. 👌

**Review by Full-lifey — APPROVED** [review]

**Comment by nortonprojects** [comment]
> I feel like the pre push hook might be too much. I'm thinking of scenarios like resolving a package.json merge conflict or something small. Having to wait for the whole test suite to run could be annoying when I've already just ran it locally. They'll take longer and longer to run too as the project grows. We can just do git push --iforget though so it's not a huge deal. I'm down for trying it.
> 
> The GitHub action is perfect though.

That's a good point. Maybe I can configure it to only run tests if relevant files have changed since the last push or something like that. That's the sort of noise I was trying to avoid having it be a pre-push instead of pre-commit, after all.

**Comment by nortonprojects** [comment]
Looks like Jest has a `--changedSince` option - I'll change it to only run files changed locally, then run the full suite in the action

**Review by markacianfrani — APPROVED** [review]

**Review by Cheshire89 — APPROVED** [review]

**Pull request merged**